src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/utils.tsx

```diff
@@ -0,0 +1,480 @@
+
+    const configList = [deploymentTemplateData, ...cmData, ...secretData]
+
+    const navList = [
+        {
+            title: deploymentTemplateData.title,
+            hasDiff: deploymentTemplateData.hasDiff,
+            href: `${generatePath(path, { ...params, resourceType: EnvResourceType.DeploymentTemplate })}${search}`,
+            onClick: () => {
+                const element = document.getElementById(deploymentTemplateData.id)
+                element?.scrollIntoView({
+                    behavior: 'smooth',
+                })
+            },
+        },
+    ]
+
+    const collapsibleNavList = [
+        {
+            header: 'Config Maps',
+            id: EnvResourceType.ConfigMap,
+            items: cmData.map(({ title, hasDiff, id }) => ({
+                title,
+                hasDiff,
+                href: `${generatePath(path, { ...params, resourceType: EnvResourceType.ConfigMap, resourceName: title })}${search}`,
+                onClick: () => {
+                    const element = document.getElementById(id)
+                    element?.scrollIntoView({
+                        behavior: 'smooth',
+                    })
+                },
+            })),
+            noItemsText: 'No configmaps',
+        },
+        {
+            header: 'Secrets',
+            id: EnvResourceType.Secret,
+            items: secretData.map(({ title, hasDiff, id }) => ({
+                title,
+                hasDiff,
+                href: `${generatePath(path, { ...params, resourceType: EnvResourceType.Secret, resourceName: title })}${search}`,
+                onClick: () => {
+                    const element = document.getElementById(id)
+                    element?.scrollIntoView({
+                        behavior: 'smooth',
+                    })
+                },
+            })),
+            noItemsText: 'No secrets',
+        },
+    ]
+
+    return {
+        configList,
+        collapsibleNavList,
+        navList,
+    }
+}
+
+export const getPreviousDeploymentOptionValue = (identifierId: number, pipelineId?: number) => {
+    if (identifierId && pipelineId) {
+        return `${AppEnvDeploymentConfigType.PREVIOUS_DEPLOYMENTS}-${identifierId}-${pipelineId}`
+    }
+    if (identifierId) {
+        return `${AppEnvDeploymentConfigType.DEFAULT_VERSION}-${identifierId}`
+    }
+    return null
+}
+
+export const getPreviousDeploymentValue = (value: string) => {
+    const valueSplit = value.split('-')
+    if (valueSplit[0] === AppEnvDeploymentConfigType.PREVIOUS_DEPLOYMENTS) {
+        return {
+            configType: AppEnvDeploymentConfigType.PREVIOUS_DEPLOYMENTS,
+            identifierId: +valueSplit[1],
+            pipelineId: +valueSplit[2],
+        }
+    }
+
+    return null
+}
+
+export const getEnvironmentIdByEnvironmentName = (environments: EnvironmentOptionType[], name: string) =>
+    environments.find(({ name: _name }) => name === _name)?.id || -1
```

Comment:
FYI, the BE team uses -1 as a placeholder for All Environments

Reply:
in this case it is being used for base configurations

Reply:
use constant for the same then pls